File: no_nonesense_rust/no-nonsense-c-source-code/12-performance/slow.c
#include <stdio.h>

int square(int n)
{
    return n * n;
}

int main()
{
    int n = 1;
    for (long i=0; i<3000000000; i++)
    {
        int square_result = square(n);
        n += square_result;
    }
    printf("Value: %d\n", n);
}

File: no_nonesense_rust/no-nonsense-rust-source-code/12-performance/slow.rs
fn square(n: i32) -> i32 {
    n * n
}

fn main() {
    let mut n: i32 = 1;
    for _ in 0..3_000_000_000u64 {
        let square_result = square(n);
        n = n.wrapping_add(square_result);
    }
    println!("Value: {}", n);
}
